transport/pool.go

```diff
@@ -105,6 +106,16 @@ type PoolRefiller struct {
 	active int
 }
 
+func changePort(addr, port *string) error {
```

Comment:
I would consider getting rid of pointers (it is not clear to me why are they needed here) and just return new string alongside with potential error.